app/Models/Device.php

```diff
@@ -330,7 +331,20 @@ public function setLocation($location_text)
 
         $this->location_id = null;
         if ($location_text) {
-            $location = Location::firstOrCreate(['location' => $location_text]);
+            $location_data = ['location' => $location_text];
+
+            if (\LibreNMS\Config::get('parse_dns_location_record')) {
+                $r = new Dns();
+                $ret = $r->getRecord($this->hostname, 'LOC');
+                $dns_record = $ret->answer;
```

Comment:
this doesnt work, you already return `$ret->answer` in the other class.

Reply:
ooops